### PR TITLE
fix the output formatting of table mode

### PR DESCRIPTION
**Pull request by rohoog**
See issue #29 

Sorry I'm not familiar with this ~more pythonic~ .format method; I used the trusty old printf style 'interpolation' instead.

### CODE REVIEW

**Comment by rohoog** [comment]
Looks like the failures are due to upper/lower case hex. This should not be an issue, but I will change %x to %X to get the same casing as before.

**Comment by Nicoretti** [comment]
Hi @rohoog,

thanks for creating a PR to address #29, really appreciate it. 
A question from my side, could you add a regression test for the issue at hand?
(So we can be sure the behavior will stay correct in the future)

You can have a look at `cli_test.py` to get an idea how such a test could look like.

Regarding the python format string vs printf style, I rather would like to keep the format string one, but I will be happy to help you out there if you want me to. In case you want to dig into it by yourself first these links may help:

* [format string syntax](https://docs.python.org/3/library/string.html#format-string-syntax)
* [format spec](https://docs.python.org/3/library/string.html#formatspec)

At first glance I think the alignment within the format string is the issue `>`

best
Nico

**Comment by rohoog** [comment]
I am not such a python guru as you might think...
The way to test such things is to use the table exporting, put it in a file (or pipe) and read it back in as the contents of the _lookup of the TableBasedCrcRegister and compare the result of CRC remainder from the built-in generated table (which works fine) and the remainder using the table read back from the file. This way you can easily cover all cases (although one case is enough to detect the bug).

**Comment by Nicoretti** [comment]
I don't think covering all cases is needed, but having the  use case which exposed the bug would be nice to have.
I think this can be done to do a similar test like [this](https://github.com/Nicoretti/crc/blob/1a79274fdaf82bea03ead21bbb72bd5a8a71d928/cli_tests.py#L30) by just adjusting the `expected_output` and the `argv` which
have been used.

> I am not such a python guru as you might think...
No problem I will guide you if you are interested in getting your PR integrated.
If you feel overwhelmed or just want to get the issue out of the way I recommend you fix it in your fork and use that one for 
now. I try to address the issue as soon as I can make some time for it.

**Comment by rohoog** [comment]
Could it even be that the expected output for the crc8 poly 0x1D is actually wrong? So that just fixing that one will 'test' it properly? I see no values _starting_ with 0x0 (except for 0x00), while there should definitely be some.
I mean just inserting the (now correctly) outputted table for that polynomial is sufficient. I think it is a bit /wrong/ to test like that... but that is another discussion.

**Comment by rohoog** [comment]
OK done that as well.

File: crc.py [hunk 1]
```diff
@@ -504,7 +504,7 @@ def table(args):
     width = args.width
     polynomial = args.polynomial
     lookup_table = create_lookup_table(width, polynomial)
-    template = '0x{{:0<0{digits}X}}'.format(digits=width // 4)
+    template = '0x{{:0{}X}}'.format((width+3) // 4)
```
Comment (Nicoretti) [inline crc.py:507]: why adding the `+3`? Currently only byte size widths are supported (may should be added to the docs)

**Comment by rohoog** [comment]
Implementation result driven testing instead of test driven development.... 😕

File: crc.py (hunk 1 above)
Comment (rohoog) [reply crc.py:507]: The +3 is when the bitlength of the polynomial is not a multiple of 4.

**Comment by Nicoretti** [comment]
> Implementation result driven testing instead of test driven development.... confused

I need to hit the bed now. As soon as I can I will validate that outputted tables are correct.
If we validate that the output is correct, then add it and keep them as test I rather would say it's a 
"snapshot" based "integration" test. Similar to what [cram](https://bitheap.org/cram/) and [prysk](https://www.prysk.net/) are doing it.

If we only would add without checking I agree it would not really a "test".

File: crc.py (hunk 1 above)
Comment (rohoog) [reply crc.py:507]: Under isssue #26, it is mentioned that CRC9 works. I can imagine that CRC<8 will not work for 256-entry table-based implementation.

**Comment by rohoog** [comment]
Hurray! Success!

File: cli_tests.py [hunk 2]
```diff
@@ -29,38 +29,38 @@ def test_table_subcommand_with_no_additional_arguments(self):
 
     def test_table_generation_for_width_8_and_poly_0x1D(self):
         expected_output = io.StringIO(inspect.cleandoc("""
```
Comment (Nicoretti) [inline cli_tests.py:31]: I have checked the table, looks like the one you added is correct. Apparently this test was setup incorrectly.

File: crc.py (hunk 1 above)
Comment (Nicoretti) [inline crc.py:507]:
```suggestion
    template = '0x{{:0{}X}}'.format((width + 3) // 4)
```

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**